### PR TITLE
Dark mode button

### PR DESCRIPTION
**Pull request by GraceGardner**
## Description & motivation 💭 
Added new button styles to button component

### Design Considerations 🎨 
https://www.figma.com/file/LNN3Cu8Akqb240WmJmbtsS/Holocene?node-id=3789%3A216118&mode=dev
Will confirm that this looks okay with Ash. 

### How was this tested 👻 
WILL UPDATE TESTS ONCE DESIGN CONFIRMS THESE ARE GOOD TO GO

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 
Please check story book and make sure buttons look like they do in figma designs

### Light
- [ ] primary
- [ ] secondary
- [ ] destructive
- [ ] ghost

### Dark
- [ ] primary
- [ ] secondary
- [ ] destructive
- [ ] ghost

### Issue(s) closed 
dark mode buttons

### CODE REVIEW

File: src/lib/theme/plugin.ts [hunk 1]
```diff
@@ -54,17 +63,20 @@ const dark: Partial<Variables<keyof typeof variables>> = {
 
   '--color-surface-primary': rgb(colors.black),
   '--color-surface-secondary': rgb(colors.offBlack),
+  '--color-surface-interactive-secondary': rgb(getColor('rhino', 950)),
   '--color-surface-subtle': rgb(getColor('slate', 900)),
   '--color-surface-badge': rgb(getColor('slate', 700)),
   '--color-surface-information': rgb(getColor('blue', 950)),
-  '--color-surface-warning': rgb(getColor('yellow', 950)),
+  '--color-surface-warning': rgb(getColor('red', 300)),
```
Comment (rossedfort) [inline src/lib/theme/plugin.ts:70]: I wonder if we should keep `warning` as is, and add a `danger` or `destructive` var for red-300. Maybe @stevekinney has thoughts?

File: src/lib/holocene/button.svelte [hunk 2]
```diff
@@ -33,13 +32,13 @@
...
+            'border-[1px] bg-interactive border-interactive text-inverse [&:not(:disabled):hover]:bg-interactiveHover [&:not(:disabled):hover]:shadow-focus [&:not(:disabled):hover]:shadow-indigo-600/50 [&:not(:disabled):hover]:border-white dark:[&:not(:disabled):hover]:border-black focus-visible:shadow-focus focus-visible:shadow-indigo-600/50 ',
           secondary:
-            ' surface-primary text-primary [&:not(:disabled):hover]:shadow-focus [&:not(:disabled):hover]:shadow-blue-600/50 [&:not(:disabled):hover]:bg-inverse [&:not(:disabled):hover]:text-white [&:not(:disabled):hover]:border-white focus-visible:shadow-focus focus-visible:shadow-blue-600/50 focus-visible:bg-inverse focus-visible:text-white focus-visible:border-white',
+            'border-2 border-secondary text-primary [&:not(:disabled):hover]:shadow-focus [&:not(:disabled):hover]:shadow-indigo-600 [&:not(:disabled):hover]:surface-interactive-secondary [&:not(:disabled):hover]:border-[1px] [&:not(:disabled):hover]:border-white dark:[&:not(:disabled):hover]:border-black focus-visible:shadow-focus focus-visible:shadow-indigo-600/50 focus-visible:bg-inverse focus-visible:border-white',
```
Comment (rossedfort) [inline src/lib/holocene/button.svelte:37]: I think we want `border-[1px]` or else the button grows/shrinks as you hover on/off of it.

https://github.com/temporalio/ui/assets/11775628/14c5770b-da5a-4a10-ae6e-ca6992de2bfc

File: src/lib/theme/colors.ts [hunk 3]
```diff
@@ -139,6 +139,34 @@ const palette: Readonly<Record<string, Record<Shade, HexColor>>> = {
     950: '#461B02',
     DEFAULT: '#F8A208',
   },
+  rhino: {
```
Comment (rossedfort) [inline src/lib/theme/colors.ts:142]: Based on the naming convention in this file, I think we should name this something that better describes what color it is.

File: src/lib/theme/colors.ts [hunk 4]
```diff
@@ -139,6 +139,34 @@ const palette: Readonly<Record<string, Record<Shade, HexColor>>> = {
...
+    400: '#74afe6',
+    500: '#5392de',
+    600: '#3f77d1',
+    700: '#3564c0',
+    800: '#31519c',
+    900: '#2c467c',
+    950: '#273860',
+    DEFAULT: '#273860',
+  },
+  pigeonPost: {
```
Comment (rossedfort) [inline src/lib/theme/colors.ts:156]: Same comment as above, although `pigeonPost` is 💯

**Review by rossedfort — COMMENTED** [review]
Looks good, just a few comments!

File: src/lib/theme/plugin.ts (hunk 1 above)
Comment (stevekinney) [reply src/lib/theme/plugin.ts:70]: I agree.

File: src/lib/holocene/button.svelte (hunk 2 above)
Comment (stevekinney) [reply src/lib/holocene/button.svelte:37]: Do we support `border-px`? I believe so.

File: src/lib/theme/colors.ts (hunk 3 above)
Comment (stevekinney) [reply src/lib/theme/colors.ts:142]: Or should it just be `slate`?

File: src/lib/theme/colors.ts (hunk 4 above)
Comment (stevekinney) [reply src/lib/theme/colors.ts:156]: I _do_ hate the camel-case color names though.

File: src/lib/holocene/button.svelte [hunk 5]
```diff
@@ -19,11 +19,11 @@
       'w-fit',
       'items-center',
       'justify-center',
+      'border-[1px]',
```
Comment (rossedfort) [inline src/lib/holocene/button.svelte:22]:
```suggestion
      'border',
```

**Comment by rossedfort** [comment]
I noticed the designs don't specify a shadow for hover state, only focus, but I think we can follow up on this later.
Screenshot 2024-02-22 at 11 15 04 AM

**Comment by rossedfort** [comment]
Also noticed ghost button focus is borked 
Screenshot 2024-02-22 at 11 26 56 AM

**Review by rossedfort — CHANGES_REQUESTED** [review]
Noticed ghost button focus state is broken
Screenshot 2024-02-22 at 11 26 56 AM

**Comment by GraceGardner** [comment]
> Noticed ghost button focus state is broken Screenshot 2024-02-22 at 11 26 56 AM

Oh no! Good catch. Thanks Ross!

**Review by rossedfort — APPROVED** [review]

**Pull request merged**